lib/pbench/test/functional/agent/cli/commands/conftest.py

```diff
@@ -22,7 +22,8 @@ def pbench_cfg(tmp_path, opt_pbench):
 
 
 @pytest.fixture
-def pbench_run(tmp_path):
+def pbench_run(monkeypatch, tmp_path):
+    monkeypatch.delenv("pbench_run")
     pbench_agent_d = tmp_path / "var" / "lib" / "pbench-agent"
     pbench_agent_d.mkdir(parents=True, exist_ok=True)
```

Comment:
It turns out that we have two definitions of the `pbench_run` fixture -- one here (the "inner" one) and one in `lib/pbench/test/functional/conftest.py` (the "outer" one, which is superseded by the inner one for tests in `agent/cil/commands` and below):
```
@pytest.fixture
def pbench_run(tmpdir):
    pbench_run = tmpdir / "test/var/lib/pbench-agent"
    os.makedirs(pbench_run)
    yield pbench_run
```

[`tmpdir`](https://docs.pytest.org/en/6.2.x/reference.html#tmpdir) here is a built-in fixture provided by PyTest.  It has [been replaced](https://docs.pytest.org/en/6.2.x/fixture.html#:~:text=replaced%20by%20tmp_path.) by [`tmp_path`](https://docs.pytest.org/en/6.2.x/reference.html#std-fixture-tmp_path) which does the same thing.

So, prior to this PR, the two `pbench_run()` fixtures did substantially the same thing, although the "outer" one inserts an extra level of subdirectory (`test`) above the `/var/lib`.  With this PR, the "inner" one will also remove the `pbench_run` definition from the environment.

I think it's a bad idea to have two fixtures with the same name which are nearly but not entirely identical, and I think it's a worse idea to let them diverge further in behavior.  So, I propose that we delete this "inner" one (which is currently being modified in this PR) and use its code to replace the "outer" one -- existing code will then use the "outer" one.  (We can debate whether the `test` level should be retained or removed...I would remove it, but maybe it's a good safeguard against `tmp_path` somehow returning a null value.)